pkg/syncer/util.go

```diff
@@ -270,3 +275,19 @@ func initVolumeMigrationService(ctx context.Context, metadataSyncer *metadataSyn
 	}
 	return nil
 }
+
+// isPVStaticallyProvisioned checks if the PersistentVolume has been provisioned statically.
+// Dynamically provisioned PVs have a volume attribute called 'storage.kubernetes.io/csiProvisionerIdentity'
+// in their CSI spec, which is set by external-provisioner.
+// Statically provisioned PVs don't have that attribute.
```

Comment:
Why not? Can we not create a PV with the same volume attribute? Does k8s prevent creating such a PV?

Reply:
+1

Reply:
This is the only attribute we found which can help identify between Dynamically created volume vs statically provisioned volume. 

Ideally end user should not impersonate, and create static PV with this attribute. `storage.kubernetes.io/csiProvisionerIdentity` is added by external-provisioner.

Do you have other alternative to help with identification of static volume? If not, then this is good check in my opinion.

Reply:
Why do we need to identify a static PV vs dynamic PV? From what I know, there is no definite way to know it.

Reply:
Can we query CNS and then decide whether to invoke `Create` or `Update` API? In the end thats what we are after right?

Reply:
We're already querying CNS to know if it's an existing CNS volume.
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/0f0e83f88be00c0baa85537f11ff8bd3cbdf3ace/pkg/syncer/metadatasyncer.go#L902

This check for dynamic vs static pv can be removed altogether. 
It's just that for dynamic PVs(which I think is the more common case in CSI), we'll avoid querying CNS by keeping this check. So it's an optimization for full sync.

Reply:
for performance optimization and to avoid the unnecessary call to QueryVolume each time PV is created dynamically if we have this check it is better.

Reply:
+1

Reply:
What happens if I statically create a PV with the volume attribute set? Would we still have the issue we are trying to fix? CNS cache DB is eventually consistent, so it also possible that a dynamically created volume in CNS may not be reported as CNS volume until the cache sync.

Reply:
If you create a static PV with this volume attribute, we would still run into the same issue. There's nothing stopping from user setting that attribute but it's unlikely user will do that.

Are you suggesting to remove this check altogether? @SandeepPissay